src/game_battlealgorithm.cpp

```diff
@@ -237,13 +236,7 @@ void Game_BattleAlgorithm::AlgorithmBase::PlayAnimation(bool on_source) {
 	}
 
 	if (on_source) {
```

Comment:
This `on_source` stuff was part of the old reflect logic, lets get rid of it entirely. Now that you're actually changing the target of the algo back to the source in `TargetFirst()` this hack is no longer needed.